src/llmq/commitment.cpp

```diff
@@ -30,7 +30,7 @@ CFinalCommitment::CFinalCommitment(const Consensus::LLMQParams& params, const ui
 template<typename... Types>
 void LogPrintfFinalCommitment(Types... out) {
     if (LogAcceptCategory(BCLog::LLMQ)) {
-        LogInstance().LogPrintStr(strprintf("CFinalCommitment::%s -- %s", __func__, tinyformat::format(out...)));
+        LogInstance().LogPrintStr(strprintf(" -- %s", tinyformat::format(out...)), "CFinalCommitment::" __FILE__, __func__, __LINE__);
```

Comment:
> with this new option disabled (which is the default state) the second line becomes 

for functional tests the default state "is enabled"
If you are debugging an incident on production, you can find by substring where it come from. 
Also please notice, that by default logs are very non-verbose if you haven't specified `-debug` option, but if you did specify manually  `-debug` option you can specify `-logfunctionnames` as well.

The particular case with too short unclear message is fixed. 
```
 [   msghand]   -- q[2aa24fd0e93a6d47f90a1b0c46d75581b7cc36a52cb3ce1559e339a652af9b4b] VALID
```

> file name and func name are in the wrong places
It was also a wrong source line and function name `LogPrintfFinalCommitment` - fixed too.

-------------

> Also, 19809 won't work correctly with lambdas e.g. queueAndMaybePushInv.

Indeed, it doesn't work with lambdas. So, for lambda you have only source file and source line.
The Log `LogPrint(BCLog::NET, "SendMessages -- queued inv: %s  index=%d peer=%d\n", invIn.ToString(), vInv.size(), pto->GetId());` produces:
```
 node0 2024-04-10T07:25:02.547034Z (mocktime: 2014-12-04T17:15:52Z) [   msghand] [net_processing.cpp:5306] [operator()] SendMessages -- queued inv: spork 088c66c8636eff4ab7b082578ec899f8aecf324bc2675659fcc9d001022516ad  index=0 peer=0 
```
which is indeed not beautiful due to `operator()` instead function name but in this particular case it didn't become worse, because `SendMessages` is already prefix string in log.

Please check newer version